seyren-core/src/main/java/com/seyren/core/service/notification/OpsGenieNotificationService.java

```diff
@@ -42,70 +45,87 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(OpsGenieNotificationService.class);
-	private SeyrenConfig config;
+    private SeyrenConfig config;
 
     @Inject
     public OpsGenieNotificationService(SeyrenConfig config) {
-    	this.config = config;
+        this.config = config;
     }
-    
+
     @Override
     public void sendNotification(Check check, Subscription subscription, List<Alert> alerts) throws NotificationFailedException {
-    	OpsGenieClient client = new OpsGenieClient();
-
+        OpsGenieClient client = new OpsGenieClient();
         try {
             String apiKey = subscription.getTarget();
-			if (check.getState() == AlertType.ERROR || check.getState() == AlertType.WARN) {
-		    	CreateAlertRequest request = new CreateAlertRequest();
-		    	request.setApiKey(apiKey);
-		    	request.setMessage(getMessage(check));
-		    	request.setSource("Seyren");
-		    	request.setTeams(config.getOpsGenieTeams());
-		
-		    	CreateAlertResponse response = client.alert().createAlert(request);
-		    	assert response.isSuccess();
+            if (check.getState() == AlertType.WARN || check.getState() == AlertType.ERROR) {
+                openAlert(client, apiKey, check);
             } else if (check.getState() == AlertType.OK) {
-            	com.ifountain.opsgenie.client.model.beans.Alert alert = findAlert(client, apiKey, getMessage(check));
-            
-            	CloseAlertRequest request = new CloseAlertRequest();
-            	request.setId(alert.getId());
-            	request.setApiKey(apiKey);
-            	CloseAlertResponse response = client.alert().closeAlert(request);
-            	assert response.isSuccess();
+                closeOpenAlerts(client, apiKey, check);
             }
-		} catch (OpsGenieClientException e) {
-			LOGGER.warn(
-					"Did not send notification to OpsGenie for check in state: {}",
-					check.getState(), e);
-		} catch (IOException e) {
-			LOGGER.warn(
-					"Did not send notification to OpsGenie for check in state: {}",
-					check.getState(), e);
-		} catch (ParseException e) {
-			LOGGER.warn(
-					"Did not send notification to OpsGenie for check in state: {}",
-					check.getState(), e);
+        } catch (OpsGenieClientException e) {
+            LOGGER.warn(
+                "Did not send notification to OpsGenie for check in state: {}",
+                check.getState(), e);
+        } catch (IOException e) {
+            LOGGER.warn(
+                "Did not send notification to OpsGenie for check in state: {}",
+                check.getState(), e);
+        } catch (ParseException e) {
+            LOGGER.warn(
+                "Did not send notification to OpsGenie for check in state: {}",
+                check.getState(), e);
+        }
+    }
 
-		}
+    private List<com.ifountain.opsgenie.client.model.beans.Alert> getOpenAlerts(OpsGenieClient client, String apiKey, Check check) throws OpsGenieClientException, IOException, ParseException {
```

Comment:
Any way to import this Alert class? Or is there an Alert class in the same package?